src/robotFactory.js

```diff
@@ -1,10 +1,104 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name,
+    weight,
+    coords = {
+      x: 0,
+      y: 0,
+    },
+    chipVersion
+  ) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
```

Comment:
it feels easier to define default coords like this
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
    this.chipVersion = chipVersion;
```

then you won't need the next two `if (...) {...}`